metrics/prometheus.go

```diff
@@ -147,6 +155,61 @@ func (o *prometheusMetrics) newHistogramMeter(name string, buckets []int64) Hist
 	}
 }
 
+func getIOLineValue(line string) (int64) {
+	fields := strings.Fields(line)
+	if len(fields) != 2 {
+		logger.Warn("this io file line is malformed", "err", line)
+		return 0
+	}
+	value, err := strconv.ParseInt(fields[1], 10, 64)
+	if err != nil {
+		logger.Warn("unable to parse int", "err", err)
+		return 0
+	}
+
+	return value
+}
+
+func getDiskIOData() (int64, int64, error) {
+	pid := os.Getpid()
+	ioFilePath := fmt.Sprintf("/proc/%d/io", pid)
+	file, err := os.Open(ioFilePath)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	// Parse the file line by line
+	scanner := bufio.NewScanner(file)
+	var reads, writes int64
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "syscr") {
+			reads = getIOLineValue(line)
+		} else if strings.HasPrefix(line, "syscw") {
+			writes = getIOLineValue(line)
+		}
+	}
+
+	return reads, writes, nil
+}
+
+func (o *prometheusMetrics) collectDiskIO(refresh time.Duration) {
+	for {
+		reads, writes, err := getDiskIOData()
+		if err != nil {
+			continue
+		} else {
+			readsMeter := o.GetOrCreateGaugeMeter("disk_reads")
```

Comment:
I might be wrong, but I think its usually better to use a counter for these things and keep increasing it.

Then in the graphs we can use either `$__rateInterval` or `$__interval`

I think @otherview knows more on this

Reply:
The reason for using a counter instead of a gauge is so that we can extract the exact values. 

With a gauge the values go up and down over time and the prometheus client may not capture all of the changes

Reply:
As far as I see @darrenvechain we do not have `Set` for our current counters (I could add it though)

Regarding the gauge I do not think the values will go in a way that is not handled by the `Set` there (so always up).

I agree that a counter makes more sense given that the value is always going up, however I think the final result would look like pretty much the same as of now.

Reply:
@freemanzMrojo for the counters we can just do `Add(...)`

Reply:
Ye yes my bad, makes more sense